### PR TITLE
lnrpc: remove legacy onion payload as an option for SendToRoute

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we remove the option to allow the caller to use the
legacy onion payload. The new payload is much more flexible and
efficient, so there's really no reason to still use it, other than for
backwards compatibility tests. Our existing tests that exercise the
legacy feature uses a build tag, which forces nodes to not advertise the
new payload format, which then forces path finding to include the legacy
payload, so we can be confident that route is still being tested.

The existence of this option (which actually makes the TLV payload
opt-in for `SendToRoute` users) makes it harder to remove it from the
protocol all together. With this PR, we take a step forward to allowing
such a change which is being tracked on the spec level at:
https://github.com/lightning/bolts/pull/962.

### CODE REVIEW

**Comment by Roasbeef** [comment]
This change may break compilation for statically compiled projects that check in or interact with the protos, so we may want to be a bit more soft handed by just ignoring that field. The main upside of this approach is that stuff will break once projects that care about this feature (hopefully!) update to test out the upcoming 0.15.0 release candidate.

File: lnrpc/lightning.proto [hunk 1]
```diff
@@ -2755,13 +2755,6 @@ message Hop {
     */
     string pub_key = 8;
 
-    /*
-    If set to true, then this hop will be encoded using the new variable length
-    TLV format. Note that if any custom tlv_records below are specified, then
-    this field MUST be set to true for them to be encoded properly.
-    */
-    bool tlv_payload = 9;
```
Comment (guggero) [inline lnrpc/lightning.proto:2763]: Set this to `reserved` instead and add a comment that it's a removed field and that the number shouldn't be re-used?

**Comment by Roasbeef** [comment]
Modifying this PR to just stop reading the field, then we can remove it in 0.16 or earlier.

**Review by guggero — APPROVED** [review]
Minimal diff looks good to me. Should we at least add the `[deprecated = true]` modifier (and maybe a comment) to the proto file so it's at least documented in the API as well?

**Comment by Roasbeef** [comment]
> Minimal diff looks good to me. 

Good idea, will add that!

**Pull request merged**